### ISSUE TITLE
NullPointerException: com.braintreepayments.api.TLSCertificatePinning.getCertInputStream (TLSCertificatePinning.java:719)

### ISSUE THREAD

**Issue by kimrtaylor** [braintree/braintree_android#1020]
### Braintree SDK Version

4.45.0

### Environment

Production

### Android Version & Device

Multiple

### Braintree dependencies

com.braintreepayments.api:card:4.45.0

### Describe the bug

We're seeing a new crash affecting some of our customers, since updating from 4.44.0 to 4.45.0. It occurs during initialisation of BraintreeClient

### To reproduce

```
Fatal Exception: java.lang.NullPointerException
Attempt to invoke virtual method 'byte[] java.lang.String.getBytes(java.nio.charset.Charset)' on a null object reference
com.braintreepayments.api.TLSCertificatePinning.getCertInputStream (TLSCertificatePinning.java:719)
com.braintreepayments.api.BraintreeHttpClient$Companion.createDefaultHttpClient (BraintreeHttpClient.java:178)
com.braintreepayments.api.BraintreeHttpClient$Companion.access$createDefaultHttpClient (BraintreeHttpClient.java:172)
com.braintreepayments.api.BraintreeHttpClient.<init> (BraintreeHttpClient.kt:12)
com.braintreepayments.api.AnalyticsClient.<init> (AnalyticsClient.kt:25)
com.braintreepayments.api.BraintreeClientParams.<init> (BraintreeClientParams.kt:14)
com.braintreepayments.api.BraintreeClientParams.<init> (BraintreeClientParams.kt:22)
com.braintreepayments.api.BraintreeClient.<init> (BraintreeClient.kt:70)
com.braintreepayments.api.BraintreeClient.<init> (BraintreeClient.kt:80)
```

### Expected behavior

exception handled

**Comment by sshropshire** [comment]
Hey @kimrtaylor thanks for using the Braintree SDK for Android. I'd like to help pinpoint the issue and hopefully find a fix.

One thing that seems odd to me is the stack trace mentions the exception originating form `TLSCertificatePinning.java` in our SDK. [TLSCetificatePinning](https://github.com/braintree/braintree_android/blob/main/BraintreeCore/src/main/java/com/braintreepayments/api/TLSCertificatePinning.kt) in our project is written in Kotlin.

Can you help me to answer a few questions:

- Have you been able to reproduce the issue?
- Is your app native Android or React Native?
- Is there any additional information you can provide about the devices that experience this issue (e.g. Android version, Device Manufacturer / Model)?

**Comment by sshropshire** [comment]
Hey @kimrtaylor @nick-titov-sw checking in. Is there any additional information available for this crash?

**Comment by sshropshire** [comment]
Thanks @nick-titov-sw. It's interesting the portion of the code that throws the NPE is virtually the same for [TLSCertificatePinning.kt in v4.44.0](https://github.com/braintree/braintree_android/blob/f97b8b0a4a8a8d4a8af81b9e2ca685eadad06ba0/BraintreeCore/src/main/java/com/braintreepayments/api/TLSCertificatePinning.kt#L92). We did update the SDK's pinned certificates, but we only changed the string contents of the `CERTIFICATES` property. Does your app have any similar crashes with the 4.44.0 version?

**Comment by sshropshire** [comment]
@nick-titov-sw going to try this to see if it helps resolve: https://github.com/braintree/braintree_android/pull/1036.

**Comment by sshropshire** [comment]
@all we've provided a potential fix for this issue in our latest [4.48.0](https://github.com/braintree/braintree_android/blob/main/CHANGELOG.md#4480-2024-07-02) version. Please upgrade and feel free to re-open this issue if the crashes persist.
